regex-syntax/src/lib.rs

```diff
@@ -105,6 +105,21 @@ done automatically in the `regex` crate.
 
 #![deny(missing_docs)]
 
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(all(feature = "alloc", not(feature = "std")), feature(alloc))]
+#![cfg_attr(all(feature = "alloc", not(feature = "std")), feature(slice_concat_ext))]
+
+#[cfg(test)]
+extern crate std as std_test;
```

Comment:
Why do this? If `std` is required for testing, you can just use regular includes from `std`. If you want to test where `std` is not available this will fail. It's possible to run many tests without `std` anyway, though might take some more work.

Reply:
The idea here is that we want to be able to run our tests against the library compiled in either mode as a way of confirming that there is no accidental behavioral differences.  

E.G. `cargo test --no-default-features --features "alloc"`

IIUIC, because we (now) do `extern crate core as std` in the main library when in no_std mode (to reduce code diffs), we need to import `std` under some other name for use in the tests so as not to conflict with the core-renamed-as-std crate being used by the library.

Reply:
I don't even know that you do need `std` for most of the testing. Have a look at the Rand lib code; we allow many tests to run with `cargo test --no-default-features` without re-importing `std`. But it's possible there are other parts of `std` you do require.

Reply:
Yes, I agree that we *could* make a decent chunk of the tests run without access to the standard library. 

That said, the main goal is testing the library code. AFAICT that library code can be compiled in no_std mode and exercised regardless of whether the test code itself has access to standard library.  As as we're getting the library code exercised already, I'm not clear on what value add there would be to go to the effort of triaging and modifying the test code to be able to run without the standard lib.